physics/continuous_trajectory_body.hpp

```diff
@@ -131,6 +131,66 @@ Status ContinuousTrajectory<Frame>::Append(
   return status;
 }
 
+template<typename Frame>
+Status ContinuousTrajectory<Frame>::Prepend(
+    ContinuousTrajectory&& trajectory) {
+  Status status;
+  absl::MutexLock l1(&lock_);
+  absl::MutexLock l2(&trajectory.lock_);
+
+  CHECK_EQ(step_, trajectory.step_);
+  CHECK_EQ(tolerance_, trajectory.tolerance_);
+
+  if (trajectory.polynomials_.empty()) {
+    // Nothing to do.
+    return Status::OK;
+  } else if (polynomials_.empty()) {
+    // All the data comes from |trajectory|.  This must set all the fields of
+    // this object that are not set at construction.
+    adjusted_tolerance_ = trajectory.adjusted_tolerance_;
+    is_unstable_ = trajectory.is_unstable_;
+    degree_ = trajectory.degree_;
+    degree_age_ = trajectory.degree_age_;
+    polynomials_ = std::move(trajectory.polynomials_);
+    last_accessed_polynomial_ = trajectory.last_accessed_polynomial_;
+    first_time_ = trajectory.first_time_;
+    last_points_ = trajectory.last_points_;
+  } else {
+    // The polynomials must be aligned, but we are lenient because it's
+    // conceivable that small differences would occur when moving a save from
+    // one machine to another.  Because of this, we cannot check that the
+    // trajectories are "continuous" at the junction, whatever that means.
```

Comment:
Should we really expect such a thing? The treatment of time with fixed steps (and continuous trajectories, which are based on fixed steps) uses only basic operations (and as we have remarked elsewhere, is often exact in practice).

Continuity in space is hopeless without correctly-rounded elementary functions, but that is another can of worms.

I would favour requiring exactness for simplicity and easier spotting of bugs, and then testing across machines with different numerical libraries (we have those).